### PR TITLE
test: use getfenv to simplify loading module

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Comment by zeertzjq** [comment]
What is the point of this?

**Comment by dundargoc** [comment]
Justin asked: https://github.com/neovim/neovim/pull/21010#discussion_r1020906114

**Comment by dundargoc** [comment]
I'm guessing while it is possible it's not very useful. Closing.

**Comment by dundargoc** [comment]
> But this assumes the two approaches are equivalent.

It has the same behavior so I assume so.

File: test/functional/api/buffer_updates_spec.lua [hunk 1]
```diff
@@ -1,4 +1,4 @@
-local helpers = require('test.functional.helpers')(after_each)
+local helpers = require('test.functional.helpers')()
```
Comment (justinmk) [inline test/functional/api/buffer_updates_spec.lua:1]: If we still need the `()` at the end, that's still kind of a hack. Do we need that?

File: test/functional/helpers.lua [hunk 2]
```diff
@@ -959,7 +959,8 @@ end
 
 module = global_helpers.tbl_extend('error', module, global_helpers)
 
-return function(after_each)
+return function()
+  local after_each = getfenv(2).after_each
```
Comment (justinmk) [inline test/functional/helpers.lua:963]: Looks like we can't avoid needing to run this function for every test file. So avoiding the argument doesn't really make things easier, and has the disadvantage of being less explicit.

Comment (dundargoc) [reply test/functional/helpers.lua:963]: I don't understand your comment. Do you want an event handler or not?

Comment (dundargoc) [reply test/functional/helpers.lua:963]: OK, then it sounds like there's no way to move forward with this PR then.

**Pull request closed without merging**